arangod/Pregel/GraphStore.cpp

```diff
@@ -534,11 +494,14 @@ void GraphStore<V, E>::loadEdges(
 
           std::string_view toValue =
               covering.at(info.coveringPosition()).stringView();
-          size_t space = toValue.size();
-          allocateSpace(space);
-          Edge<E>* edge = edgeBuff->appendElement();
-          buildEdge(edge, toValue);
-          return true;
+
+          auto newEdge = buildEdge(toValue);
+          if (newEdge.has_value()) {
+            vertex.emplaceEdge(std::move(*newEdge));
+            return true;
+          } else {
+            return false;
+          }
```

Comment:
Readability can be improved if you first check for the errorcase. Then you return early and don't need an else afterwards